diff -rupN mutter-3.8.3.orig/src/core/window.c mutter-3.8.3/src/core/window.c
--- mutter-3.8.3.orig/src/core/window.c	2013-06-08 00:23:05.000000000 +0600
+++ mutter-3.8.3/src/core/window.c	2013-10-23 10:34:50.688468200 +0600
@@ -2808,7 +2808,8 @@ window_state_on_map (MetaWindow *window,
     case META_WINDOW_NORMAL:
     case META_WINDOW_DIALOG:
     case META_WINDOW_MODAL_DIALOG:
-      /* The default is correct for these */
+      *takes_focus = TRUE;
+      *places_on_top = TRUE;
       break;
     }
 }
